controllers/harborconfiguration_controller.go

```diff
@@ -309,68 +297,20 @@ func getHarborSecret(ctx context.Context, clientSet *kubernetes.Clientset, harbo
 	if err != nil {
 		return "", err
 	}
-	return string(passwordSecret.Data["secret"]), nil
+	passwords := passwordSecret.Data
+	for key, value := range passwords {
+		if key == "secret" && string(value) != "" {
+			return string(value), nil
```

Comment:
Golang maps have a nicety to check if the key exists:
```suggestion
password, ok := passwordSecret.Data["secret"]
if !ok {
   return "", errors.New(`no key "secret" found`)
}
return password, nil
```
https://go.dev/blog/maps